demosplan/DemosPlanCoreBundle/Logic/OwnsProcedureConditionFactory.php

```diff
@@ -13,120 +13,154 @@
+            $user = $this->userOrProcedure;
+
+            return $user->hasAnyOfRoles($relevantRoles, $customer)
+                ? [$this->conditionFactory->true()]
+                : [$this->conditionFactory->false()];
+        }
+
+        $procedure = $this->userOrProcedure;
+
+        if (null !== $procedure->getOrgaId()) {
             $this->logger->debug('Permissions: Check whether orga owns procedure');
-            // Fachplaner-Admin GLAUTH Kommune oder Fachplaner SB
-
-            // Fachplaner admin oder Fachplaner Sachbearbeiter oder Plattform-Admin oder AHB-Admin
-
-            $ownsOrgaRoleCondition = $this->conditionFactory->propertyHasAnyOfValues(
-                [
-                    Role::CUSTOMER_MASTER_USER,
-                    Role::PLANNING_AGENCY_ADMIN,
-                    Role::PLANNING_AGENCY_WORKER,
-                    Role::HEARING_AUTHORITY_ADMIN,
-                    Role::HEARING_AUTHORITY_WORKER,
-                ],
-                ['roleInCustomers', 'role', 'code']
-            );
+            $ownsOrgaRoleCondition = [
+                $this->conditionFactory->propertyHasAnyOfValues($relevantRoles, ['roleInCustomers', 'role', 'code']),
+                $this->isUserInCustomer($customer),
+            ];
+        } else {
+            $ownsOrgaRoleCondition = [$this->conditionFactory->false()];
         }
 
         return $ownsOrgaRoleCondition;
     }
 
     /**
-     * The user must have the {@link Role::PRIVATE_PLANNING_AGENCY} role.
+     * The user must have the {@link RoleInterface::PRIVATE_PLANNING_AGENCY} role.
      *
-     * @return FunctionInterface<bool>
+     * @return list<FunctionInterface<bool>>
      */
-    public function hasPlanningAgencyRole(): FunctionInterface
+    public function hasPlanningAgencyRole(Customer $customer): array
     {
-        $planningAgencyOwnsProcedure = $this->conditionFactory->false();
+        $relevantRole = RoleInterface::PRIVATE_PLANNING_AGENCY;
+
+        if ($this->userOrProcedure instanceof User) {
+            return $this->userOrProcedure->hasRole($relevantRole, $customer)
+                ? [$this->conditionFactory->true()]
+                : [$this->conditionFactory->false()];
+        }
+
+        $procedure = $this->userOrProcedure;
 
-        if (0 < count($this->procedure->getPlanningOfficesIds())) {
+        if (0 < count($procedure->getPlanningOfficesIds())) {
             $this->logger->debug('Procedure has PlanningOffices');
 
             // ist es ein PLanungsbüro?
-            $planningAgencyOwnsProcedure = $this->conditionFactory->propertyHasValue(
-                Role::PRIVATE_PLANNING_AGENCY,
-                ['roleInCustomers', 'role', 'code']
-            );
+            $planningAgencyOwnsProcedure = [
+                $this->conditionFactory->propertyHasValue($relevantRole, ['roleInCustomers', 'role', 'code']),
+                $this->isUserInCustomer($customer),
+            ];
+        } else {
```

Comment:
you can avoid this else

Reply:
https://github.com/demos-europe/demosplan-core/pull/2662/files#r1478637156